### PR TITLE
Add logging levels to ivy

### PR DESCRIPTION
**Pull request by sladyn98**
This pull request introduces a global logging_mode to Ivy, aiming to control the amount of logging done across various execution paths. The logging modes follow Python’s default logging module conventions i.e. DEBUG, INFO, WARNING, ERROR.This enhancement improves the debugging and development process by controlling the verbosity of logging. Users can now adjust the logging mode to match their specific requirements, gaining insights into the execution paths and identifying potential issues more effectively.

### CODE REVIEW

**Comment by sladyn98** [comment]
@soma2000-lang 

Could you help me on which model would we like to test this.

**Comment by soma2000-lang** [comment]
@sladyn98 We need to add some logging modes temporarily to see if it works.

**Comment by sladyn98** [comment]
@soma2000-lang I added a temporary logging statement to set the logging methods, is there a way to verify it.

**Review by vedpatwardhan — COMMENTED** [review]
I assume the automated review request wasn't intended given that it seems to be a WIP, feel free to request a review whenever you're ready. Thanks @sladyn98 @soma2000-lang 🙂

**Comment by sladyn98** [comment]
I am ready @vedpatwardhan

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Awesome, thanks for working on this @sladyn98 and @soma2000-lang 💪
I think it would be more consistent if we define these modes in the `ivy/__init__.py`. , similar to the `warning_level`.
Also, could you please remove the setting and unsetting of the mode from `handle_nans`? Probably that's accidentally left to be removed.
Also, let's rename `test_logging.py` to `test_modes.py`, eventually we'd add the tests for all [operating modes](https://unify.ai/docs/ivy/overview/deep_dive/operating_modes.html) in that file 🙂

**Comment by soma2000-lang** [comment]
Hey @sladyn98 @vedpatwardhan ,I would like to add tests for all the operating modes.Will open a seperate PR  for this?

**Comment by vedpatwardhan** [comment]
> Hey @sladyn98 @vedpatwardhan ,I would like to add tests for all the operating modes.Will open a seperate PR for this?

Hey @soma2000-lang, thanks for showing interest! Yeah feel free to create a new PR for adding the tests for all operating modes 🚀

**Comment by sladyn98** [comment]
@vedpatwardhan With this PR do we need to add more testing ? Is there any way to verfiy the logging works ?

**Comment by vedpatwardhan** [comment]
> @vedpatwardhan With this PR do we need to add more testing ? Is there any way to verfiy the logging works ?

Hey @sladyn98, the tests for the other modes would be added in a different PR which isn't related to this one. I think given that we're wrapping the `logging` module, we wouldn't really need to verify as much. But it would be good to just add a few logging statements temporarily and seting it with different modes just to be sure. Those logging statements would eventually be removed of course!
Once that's done, we can get the PR merged! Could you please also remove the mode setting logic [here](https://github.com/unifyai/ivy/pull/18020/files#diff-e507595f1a13b0ef477a10a0399fa210f0708466a9f19197a27dda65986bbf09R1359)? Thanks 😄

**Comment by vedpatwardhan** [comment]
ivy-gardener

**Comment by sladyn98** [comment]
@vedpatwardhan Thanks, I removed the setting statements, I was thinking that since the codebase has enough of logging statements we could instead test if the logging levels are set using the setting and getting methods. or after the tests have been merged

**Comment by vedpatwardhan** [comment]
> @vedpatwardhan Thanks, I removed the setting statements, I was thinking that since the codebase has enough of logging statements we could instead test if the logging levels are set using the setting and getting methods. or after the tests have been merged

Yep sounds perfect! I think the tests in `test_logging.py` currently are sufficient in that case, feel free to get the PR merged. Thanks @sladyn98 😄

File: ivy/func_wrapper.py [hunk 1]
```diff
@@ -1365,6 +1365,7 @@ def _handle_nans(*args, **kwargs):
             elif nan_policy == "warns":
                 logging.warning("Nans are present in the input.")
 
+
```
Comment (CatB1t) [inline ivy/func_wrapper.py:1368]: Let's revert this line here. lint-fixes shouldn't be part of this PR.

File: ivy/utils/backend/sub_backend_handler.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 _sub_backend_dict = dict()
 _backend_to_sub_backends_dict = dict()
 
-
```
Comment (CatB1t) [inline ivy/utils/backend/sub_backend_handler.py:15]: Let's revert this line here. lint-fixes shouldn't be part of this PR.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy/utils/backend/sub_backend_handler.py (hunk 2 above)
Comment (sladyn98) [reply ivy/utils/backend/sub_backend_handler.py:15]: Fixed

File: ivy/func_wrapper.py (hunk 1 above)
Comment (sladyn98) [reply ivy/func_wrapper.py:1410]: Fixed
